atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java

```diff
@@ -181,12 +181,14 @@ default String getKeyspaceOrThrow() {
     Optional<SslConfiguration> sslConfiguration();
 
     /**
-     * An object which implements the logic behind CQL client resource management. Default implementation creates a new
-     * one per CKVS.
+     * An object which implements the logic behind CQL communication resource management. Default factory object creates
+     * a new {@link com.palantir.atlasdb.keyvalue.api.AsyncKeyValueService} with new session and thread pool every time.
+     * For smarter resource management this option should be programmatically set for client specific resource
+     * management.
      */
     @Value.Default
-    default CqlClientFactory cqlClientFactory() {
-        return CqlClientFactoryImpl.INSTANCE;
+    default CassandraAsyncKeyValueServiceFactory asyncKeyValueServiceFactory() {
```

Comment:
It feels like this should be annotated `@JsonIgnore`: serializing a `CassandraAsyncKeyValueServiceFactory` is probably not a reasonable thing to do, and this class has always been annotated with `JsonSerialize` so I'm not sure I'd like to deal with the ramifications of preventing that from happening.

Reply:
Added.